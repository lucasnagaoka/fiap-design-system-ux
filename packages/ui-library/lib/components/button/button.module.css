.button {
  background: greenyellow;
  color: black;
  border: 1px solid black;
  padding: 10px 20px;
}
