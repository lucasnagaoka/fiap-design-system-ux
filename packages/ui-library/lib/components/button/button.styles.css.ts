import { style } from '@vanilla-extract/css';
import { FIAP_DS_COLORS } from '../../design-tokens/colors';
import { FIAP_DS_SPACING } from '../../design-tokens/spacing';
import { FIAP_DS_TYPOGRAPHY } from '../../design-tokens/typography';

const button = style({
  padding: FIAP_DS_SPACING.s4,
  borderRadius: FIAP_DS_SPACING.s2,
  width: '100%',
  minWidth: FIAP_DS_SPACING.s6,
  maxWidth: FIAP_DS_SPACING.s10 * 2,
  fontSize: FIAP_DS_TYPOGRAPHY.p['font-size'],
  lineHeight: FIAP_DS_TYPOGRAPHY.button['line-height'],
  fontWeight: FIAP_DS_TYPOGRAPHY.button.weight,
  textAlign: 'center',
  borderWidth: 0,
});

const buttonPrimary = style({
  backgroundColor: FIAP_DS_COLORS.primary[900],
  color: FIAP_DS_COLORS.white[100],
});

const buttonSecondary = style({
  backgroundColor: FIAP_DS_COLORS.white[100],
  color: FIAP_DS_COLORS.primary[900],
  borderColor: FIAP_DS_COLORS.primary[900],
  borderWidth: '1px',
  borderStyle: 'solid',
});

const buttonTertiary = style({
  backgroundColor: FIAP_DS_COLORS.white[100],
  color: FIAP_DS_COLORS.black[25],
  borderColor: FIAP_DS_COLORS.black[25],
  borderWidth: '1px',
  borderStyle: 'solid',
});

export { button, buttonPrimary, buttonSecondary, buttonTertiary };
