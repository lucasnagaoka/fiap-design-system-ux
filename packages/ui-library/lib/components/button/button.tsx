import { clsx } from 'clsx';
import { fontText } from '../../styles/font.css';
import {
  button,
  buttonPrimary,
  buttonSecondary,
  buttonTertiary,
} from './button.styles.css';

import type { JSX } from 'react';
import type { ButtonProps, ButtonVariant } from './button.types.ts';

export function Button({
  children,
  variant = 'button',
  ...other
}: ButtonProps): JSX.Element {
  function getVariantClass(variant: ButtonVariant) {
    switch (variant) {
      case 'primary':
        return buttonPrimary;
      case 'secondary':
        return buttonSecondary;
      case 'tertiary':
        return buttonTertiary;
      default:
        return button;
    }
  }

  return (
    <button
      className={clsx([fontText, button, getVariantClass(variant)])}
      {...other}
    >
      {children}
    </button>
  );
}

Button.displayName = 'Button';
