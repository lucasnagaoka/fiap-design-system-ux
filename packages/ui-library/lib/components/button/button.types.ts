export type ButtonVariant = 'primary' | 'secondary' | 'tertiary' | 'button';

export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  children: React.ReactNode;
  variant?: ButtonVariant;
};
