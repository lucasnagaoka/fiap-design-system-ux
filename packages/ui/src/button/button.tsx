import type { JSX } from 'react';
import type { ButtonProps } from './button.types';

export function Button({ children, ...other }: ButtonProps): JSX.Element {
  return (
    <button type="button" {...other}>
      {children}
    </button>
  );
}

Button.displayName = 'Button';
