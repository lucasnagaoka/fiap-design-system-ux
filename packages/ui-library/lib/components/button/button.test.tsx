import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { Button } from './button';
import * as styles from './button.styles.css.ts';

describe('Button component', () => {
  it('Should render an primary button with Hello text', () => {
    render(<Button children='Hello' variant='primary' />);
    const button = screen.getByRole('button', { name: /hello/i });

    expect(button.classList.contains(styles.button)).toBeTruthy();
    expect(button.classList.contains(styles.buttonPrimary)).toBeTruthy();
  });

  it('Should render an secondary button with Hello text', () => {
    render(<Button children='Hello' variant='secondary' />);
    const button = screen.getByRole('button', { name: /hello/i });

    expect(button.classList.contains(styles.button)).toBeTruthy();
    expect(button.classList.contains(styles.buttonSecondary)).toBeTruthy();
  });

  it('Should render an tertiary button with Hello text', () => {
    render(<Button children='Hello' variant='tertiary' />);

    const button = screen.getByRole('button', { name: /hello/i });

    expect(button.classList.contains(styles.button)).toBeTruthy();
    expect(button.classList.contains(styles.buttonTertiary)).toBeTruthy();
  });

  it('Should render a clickable button', () => {
    const handleClick = vi.fn();

    render(
      <Button children='Sing in' variant='primary' onClick={handleClick} />
    );
    fireEvent.click(screen.getByText('Sing in'));

    expect(handleClick).toHaveBeenCalled();
    expect(screen.getByText('Sing in')).toBeTruthy();
  });

  it('Should take a button snapshot', () => {
    const { asFragment } = render(
      <Button children='Sing up' variant='primary' />
    );

    expect(asFragment).toMatchSnapshot();
  });
});
