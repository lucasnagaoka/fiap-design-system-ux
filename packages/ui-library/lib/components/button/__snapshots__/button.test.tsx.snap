// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > Should take a button snapshot 1`] = `[Function]`;
